src/adapters/IDB.js

```diff
@@ -142,6 +142,27 @@ const cursorHandlers = {
   },
 };
 
+const transformSubObjectFilters = filtersObj => {
+  const newFilter = {};
+  const makeNestedObjectFromArr = (arr, val) => {
+    const last = arr.length - 1;
+    arr.reduce((acc, cv, i) => {
+      return i === last
+        ? (acc[cv] = val)
+        : acc.hasOwnProperty(cv)
+        ? acc[cv]
+        : (acc[cv] = {});
+    }, newFilter);
+    return newFilter;
+  };
+  for (const key in filtersObj) {
+    const keysArr = key.split(".");
+    const val = filtersObj[key];
+    makeNestedObjectFromArr(keysArr, val);
```

Comment:
It's better if you don't mix scopes (for example here `newFilter` is modified outside this loop). You should take advantage of the fact the `reduce()` returns the result.

For example:
```js
for (const key in filtersObj) {
  const keysArr = key.split(".");
  const topLevel = keysArr[0];
  const val = filtersObj[key];
  filtersObj[topLevel] = makeNestedObjectFromArr(filtersObj[topLevel], keysArr, val);
}
```